tests/scripts/ci_perf.py

```diff
@@ -0,0 +1,177 @@
+
+def run_command(runArgs, environment):
+    """ Run command specified by runArgs
+    Args:
+        runargs (str[]): list of arguments for subprocess
+        environment(str{}): dict of environment variable
+    """
+    log('RUNNING COMMAND: ' + runArgs)
+    subprocess.check_call(runArgs, env=environment)
+
+##########################################################################
+# CI jobs
+##########################################################################
+
+class jenkins_properties:
+    def __init__(self, workspace, git_branch_without_origin, git_branch, git_commit):
+        self.workspace = workspace
+        self.git_branch_without_origin = git_branch_without_origin
+        self.git_branch = git_branch
+        self.git_commit = git_commit
+
+
+class coreclr_scenarios_args:
+    def __init__(self, is_pr, arch, os, jit_name, opt_level, benchview_commit_name, bv_upload_sas_token, upload_to_benchview):
+        self.is_pr = is_pr
+        self.arch = arch
+        self.os = os
+        self.jit_name = jit_name
+        self.opt_level = opt_level
+        self.benchview_commit_name = benchview_commit_name
+        self.bv_upload_sas_token = bv_upload_sas_token
+        self.upload_to_benchview = upload_to_benchview
+
+def coreclr_scenarios(args, props):
+    architecture = args.arch
+    testEnv = ''
+    configuration = 'Release'
+    runType = 'private' if args.is_pr else 'rolling'
+    benchViewName = 'CoreCLR-Scenarios private ' + args.benchview_commit_name if args.is_pr else 'CoreCLR-Scenarios rolling ' + props.git_branch_without_origin + ' ' + props.git_commit
+    uploadString = '-uploadToBenchview' if args.upload_to_benchview else ''
+    
+    run_command('powershell -NoProfile wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe -OutFile \"' + props.workspace + '\\nuget.exe\"', os.environ)
+    if os.path.isdir(props.workspace + '\\Microsoft.BenchView.JSONFormat'):
+        shutil.rmtree(props.workspace + '\\Microsoft.BenchView.JSONFormat')
+    run_command('\"' + props.workspace + '\\nuget.exe\" install Microsoft.BenchView.JSONFormat -Source http://benchviewtestfeed.azurewebsites.net/nuget -OutputDirectory \"' + props.workspace + '\" -Prerelease -ExcludeVersion', os.environ)
+    run_command('py \"' + props.workspace + '\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"' + benchViewName + '\" --user-email \"dotnet-bot@microsoft.com\"', os.environ)
+    run_command('py \"' + props.workspace + '\\Microsoft.BenchView.JSONFormat\\tools\\build.py\" git --branch ' + props.git_branch_without_origin + ' --type ' + runType, os.environ)
+    run_command('py \"' + props.workspace + '\\Microsoft.BenchView.JSONFormat\\tools\\machinedata.py\"', os.environ)
+    run_command('cmd.exe /c set __TestIntermediateDir=int&&build.cmd ' + configuration + ' ' + architecture, os.environ)
+    run_command('tests\\runtest.cmd ' + configuration + ' ' + architecture + ' GenerateLayoutOnly', os.environ)
```

Comment:
I would add an option to build and generate layout, because the general user scenario is to run benchmarks only.

Reply:
My goal is just to replicate what perf.groovy does so that I could test. I think it would be fine if someone else wants to add more options, but I don't want to add more feature requests to my own scope of work.